### PR TITLE
Add CS, PRL, LDC and RDC access patterns with supporting config

### PR DESCRIPTION
**Pull request by aniket-modi**
Add 4 new data access patterns to the h5bench_read benchmark

### CODE REVIEW

File: h5bench_patterns/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+## Additions to H5Bench_read
+4 new patterns are added to the h5bench read benchmark:
+1. CS: Refers to the Cross Stencil data access pattern. A block of fixed sides is used to read data from hdf. This block is given a fixed stride in each dimension and data till end of file is read.
```
Comment (sbyna) [inline h5bench_patterns/README.md:3]: Minor comment: "hdf" --> "HDF5".

File: h5bench_patterns/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+## Additions to H5Bench_read
+4 new patterns are added to the h5bench read benchmark:
```
Comment (sbyna) [inline h5bench_patterns/README.md:2]: Make sure to mention that this works only with single MPI process.

File: h5bench_patterns/h5bench_read.c [hunk 3]
```diff
@@ -200,6 +200,143 @@ _set_dataspace_seq_2D(hid_t *filespace_in_out, hid_t *memspace_out, unsigned lon
     return dim_1 * dim_2;
 }
```
Comment (sbyna) [inline h5bench_patterns/h5bench_read.c:202]: Minor: Add a short description here and at the start of the remaining functions describing what this function does.

File: samples/sync-write-1d-contig-contig-read-CS.json [hunk 4]
```diff
@@ -0,0 +1,55 @@
+{
+    "mpi": {
+        "command": "mpirun",
+        "ranks": "1",
+        "configuration": "--allow-run-as-root -n 1 --oversubscribe"
```
Comment (sbyna) [inline samples/sync-write-1d-contig-contig-read-CS.json:5]: Is this specific to the environment where it was run? How would this change if we have to run on Perlmutter or some other system?

File: samples/sync-write-1d-contig-contig-read-LDC.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
+{
+    "mpi": {
+        "command": "mpirun",
+        "ranks": "4",
+        "configuration": "--allow-run-as-root -n 2 --oversubscribe"
```
Comment (sbyna) [inline samples/sync-write-1d-contig-contig-read-LDC.json:5]: System specific?

File: samples/sync-write-1d-contig-contig-read-CS.json (hunk 4 above)
Comment (aniket-modi) [reply samples/sync-write-1d-contig-contig-read-CS.json:5]: This was taken from existing sample configuration files. I took the file `sync-write-1d-contig-contig-read-full.json` as reference for the new configs. This part would be changed just like for other existing patterns on a new system.

File: h5bench_patterns/h5bench_read.c (hunk 3 above)
Comment (aniket-modi) [reply h5bench_patterns/h5bench_read.c:202]: Fixed

File: h5bench_patterns/README.md (hunk 2 above)
Comment (aniket-modi) [reply h5bench_patterns/README.md:2]: Added.

File: h5bench_patterns/README.md (hunk 1 above)
Comment (aniket-modi) [reply h5bench_patterns/README.md]: Fixed

**Review by sbyna — APPROVED** [review]

File: samples/sync-write-1d-contig-contig-read-CS.json (hunk 4 above)
Comment (jeanbez) [reply samples/sync-write-1d-contig-contig-read-CS.json:5]: Our CI should be able to handle that change when running in Perlmutter.

**Pull request merged**